### PR TITLE
PRO-2625 object fields

### PR DESCRIPTION
**Pull request by boutell**
## Summary

* Support for the object field type, per the design

## What are the specific steps to test this change?

See pro-2625 branch of `testbed`. The `topic` piece type now has an object field, with a nested sub-object field, in its "fancy fields" tab.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [separate PR coming] Related documentation has been updated
- [X] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/schema/index.js [hunk 1]
```diff
@@ -790,6 +790,7 @@ module.exports = {
     self.addFieldType({
       name: 'object',
       async convert(req, field, data, destination) {
+        data = data[field.name];
```
Comment (ValJed) [inline modules/@apostrophecms/schema/index.js:793]: Not a big deal here, but avoiding to reassign is a good thing for code maintainability.

File: modules/@apostrophecms/schema/ui/apos/components/AposInputObject.vue [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  .apos-input-object {
+    border-left: 1px solid var(--a-base-9);
+  }
+  .apos-input-wrapper {
+    margin: 20px 0 0 19px;
+  }
+  .apos-input-object ::v-deep .apos-schema .apos-field {
+    margin-bottom: 30px;
+  }
+</style>
```
Comment (ValJed) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputObject.vue:67]: No new line at end of file. We must add an eslint rule for this.

**Review by ValJed — APPROVED** [review]

**Comment by Seao** [comment]
First of all, thank you for this very expected feature @boutell  🎉

It works well, I only noted a strange case with an `object` field called `error`, all its subfields appear as invalid when it is not.

```js
error: {
  type: 'object',
  fields: {
    add: {
      child: {
        type: 'string',
        label: 'Child field'
      }
    }
  }
},
```

bug

By changing the name of the `error` field, everything works. Is it a protected field name?

**Comment by boutell** [comment]
I see your note @Seao, we will mark this as an issue for later followup, for now avoid that field name.

**Pull request merged**